includes/admin/class-wp-job-manager-admin-notices.php

```diff
@@ -636,7 +648,7 @@ private static function render_notice( $notice_id, $notice ) {
 			}
 		}
 		if ( $is_dismissible ) {
-			echo '<button type="button" class="wpjm-button is-link notice-dismiss"><span class="screen-reader-text">' . esc_html__( 'Dismiss this notice', 'wp-job-manager' ) . '</span></button>';
+			echo '<button type="button" class="wpjm-button is-link notice-dismiss wpjm-notice-dismiss"><span class="screen-reader-text">' . esc_html__( 'Dismiss this notice', 'wp-job-manager' ) . '</span></button>';
```

Comment:
Tried to make the CSS class more explicit here, adding a new `..--icon` variation: 9065fff

A few things come together here:
- WordPress core automatically adds a new X button with the `.notice-dismiss` class to dismissible notices, unless it already finds an element with that class. So it has to be present in regular notices.
- I wanted to separate the classes for the behavior and styles because in the new editor modal we need different styling. This is now `.wpjm-notice-dismiss` for the behavior (our JS attaching a click handler), and `.wpjm-notice-dismiss--icon` for the notice X style.